src/util.jl

```diff
@@ -126,7 +134,14 @@ Base.front(I::CartesianIndex) = CI(Base.front(I.I))
 
 Apply a vector function `f(i,x)` to the faces of a uniform staggered array `c`.
 """
-apply!(f,c) = @loop c[Ii] = f(last(Ii),loc(Ii)) over Ii ∈ CartesianIndices(c)
+apply!(f,c) = hasmethod(f,Tuple{Int,CartesianIndex}) ? applyV!(f,c) : applyS!(f,c)
+applyV!(f,c) = @loop c[Ii] = f(last(Ii),loc(Ii)) over Ii ∈ CartesianIndices(c)
+""" 
+    apply!(f, c)
```

Comment:
I think you need to just write one function description for both function. Here you haven't explained the difference and it seems confusing.